src/org/zaproxy/zap/extension/alert/ExtensionAlert.java

```diff
@@ -376,6 +380,23 @@ private void writeAlertToDB(Alert alert, HistoryReference ref) throws HttpMalfor
 
     }
 
+    private TreePath findAlertInTree(Alert alert) {

```

Comment:
Index 0 might be fine, if selecting the first node. How many nodes did you have? (I tested the method and it seemed to work fine, selected the expected node always).

That method is not as efficient, it traverses the whole tree (in worst case).